draft-ietf-quic-qlog-main-schema.md

```diff
@@ -244,6 +246,57 @@ schema structured as:
     * `scenario` event type
     * `marker` event type
 
+# Abstract LogFile Class {#abstract-logfile}
+
+A Log file is intended to contain a collection of events that are in some way
+related. An abstract LogFile class containing fields common to all log files is
+defined. Each concrete log file format derives from this, extending it by
+defining semantics and any custom fields.
+
+~~~ cddl
+LogFile = {
+    file_schema: text
+    qlog_serialization_format: text
+    ? title: text
+    ? description: text
+    event_schema: [+text]
+
+    ; can contain any amount of custom fields
+    * text => any
```

Comment:
You probably know this, but this should be something like `$$logfile-extensions` with CDDL

Reply:
Makes sense but then the concrete log files need to extend that right? I gave it a go, PTAL

Reply:
So turns out my suggestion of using the group socket doesn't actually work :) We need to use the CDDL "unwrapping operator" `~` instead. This will "copy and paste" all the fields from the operand struct over to this new struct. This way, we CAN define new struct names for QlogFile and QlogFileSeq :) 

So the correct CDDL definitions for the 3 structs are (note: removed `$$logfile-extensions` from `LogFile`!)

```
~~~ cddl
LogFile = {
    file_schema: text
    serialization_format: text
    ? title: text
    ? description: text
    event_schemas: [+text]
}
~~~
{: #abstract-logfile-def title="LogFile definition"}
```

```
~~~ cddl
QlogFile = {
    ~LogFile
    ? traces: [+ Trace /
                TraceError]
}
~~~
{: #qlog-file-def title="QlogFile definition"}
```

```
~~~ cddl
QlogFileSeq = {
    ~LogFile
    trace: TraceSeq
}
~~~
{: #qlog-file-seq-def title="QlogFileSeq definition"}
```


Note that we used this before in H3, but I removed those for clarity. There, I also had some clarifying text (much like we have for the sockets), like this:

> As a reminder the CDDL unwrap operator (~), see {{?RFC8610}}), copies the fields
from the referenced type (H3Parameters) into the target type directly, extending the
target with the unwrapped fields.

Maybe a good idea to add something like that in here as well, to clarify what we're doing (replacing the current text around `$$logfile-extensions`)?